### ISSUE TITLE
macOS: Build failure with Clang 14.

### ISSUE THREAD

**Issue by FtZPetruska** [libsdl-org/SDL#5846]
[Starting Clang 14](https://releases.llvm.org/14.0.0/tools/clang/docs/ReleaseNotes.html#improvements-to-clang-s-diagnostics), the warning `declaration-after-statement` is checked regardless of the standard (previously only checked when standard was set to <=c90).

This change makes build fail when targeting an Apple platform as most Objective-C files are not designed around this, having runtime availability checks before starting any work, const variables assigned mid-functions, etc.

Since the affected files are Apple-only, the warning could safely be disabled for these files as no other platform should use it and supported Apple compilers are all >=c99.

Or should the affected files be patched, preferably before Xcode 14 gets officially released ? With a patch [like this](https://github.com/libsdl-org/SDL/files/8985932/macos-declaration-after-statement.txt) I was able to get it to compile and it seemed to run example programs without any issue, but some functions get really messy, most notably `METAL_CreateRenderer` with 62 lines of variable declaration at the beginning of the function.

Testing environment:
- M1 CPU
- macOS 12.4
- Homebrew LLVM Clang 14
- CMake/Autotools
- Tested on: 787dc260a8a788a0d2319fe76f7756e0bc607e03 (HEAD) and tag `release-2.0.22`

**Comment by slouken** [comment]
Yes, I would feel more comfortable if the Objective C code used the same style of declarations as the C code. Do you want to create a PR with your patch?

**Comment by FtZPetruska** [comment]
Sure, I'll just make sure it still functions on the latest commit before submitting it.
